application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -159,8 +159,11 @@ public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
         // This method handles chatgpt's automatic response "dismiss" button
         ThreadChannel channel = event.getChannel().asThreadChannel();
         Member interactionUser = Objects.requireNonNull(event.getMember());
-        if (channel.getOwnerIdLong() != interactionUser.getIdLong()
-                && !helper.hasTagManageRole(interactionUser)) {
+        Message forumPostMessage = channel.retrieveMessageById(channel.getId()).complete();
```

Comment:
use a proper rest action chain, dont do `complete()`.
also, u have to `deferReply`, since the retrieval could take a few seocnds.

Reply:
this one took a while to work out but i think i got it right.